File: BsdUpdated/BsdUpdated/Models/Gift.cs
namespace BsdUpdated.Models
{
    public class Gift
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Cost { get; set; }
        public string Picture { get; set; }
        public int CategoryId { get; set; }
        public int DonorId { get; set; }
        public List<Card> CardsList { get; set; } 
        public string WinnerName { get; set; }
        public Donor Donor { get; set; }
        public Category Category { get; set; }


    }
}

File: BsdFinalProject-C/src/app/Models/gift.ts
import { CardModel } from "./card";

export class GiftModel {
  id!: number;
  name!: string;
  description?: string;
  cost!: number;
  picture?: string;
  categoryId!: number;
  donorId!: number; // קשר בין המתנה לתורם
  winnerName?: string;
  tickets?: CardModel[];
  donorName?: string; // שדה נוסף לשם התורם
}